pkg/volume/util/util.go

```diff
@@ -88,29 +89,34 @@ func UnmountPath(mountPath string, mounter mount.Interface) error {
 // IsNotMountPoint will be called instead of IsLikelyNotMountPoint.
 // IsNotMountPoint is more expensive but properly handles bind mounts.
 func UnmountMountPoint(mountPath string, mounter mount.Interface, extensiveMountPointCheck bool) error {
+	isStaleConnection := false
 	if pathExists, pathErr := PathExists(mountPath); pathErr != nil {
-		return fmt.Errorf("Error checking if path exists: %v", pathErr)
+		isStaleConnection = IsStaleConnection(pathErr)
+		if !isStaleConnection {
+			return fmt.Errorf("Error checking if path exists: %v", pathErr)
+		}
 	} else if !pathExists {
 		glog.Warningf("Warning: Unmount skipped because path does not exist: %v", mountPath)
 		return nil
 	}
 
 	var notMnt bool
```

Comment:
Move these two vars declaration to inner if-clause?

Reply:
do you mean notMnt and err？
but notMnt is used out of the if-clause